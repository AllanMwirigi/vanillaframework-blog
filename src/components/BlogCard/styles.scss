.blog-post-col {
  margin-bottom: 1.25rem;
}
.blog-post {
  display: flex;
  flex-direction: column;
  border-top: 3px solid purple;
  min-height: 100%;
}
.header-break {
  margin-bottom: 0.75rem;
}
.blog-card-footer {
  border-top: 1px dotted gray;
  margin-top: auto;
  padding-top: 0.5rem;
}