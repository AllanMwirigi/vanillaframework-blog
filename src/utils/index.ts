import { DateTime } from 'luxon';
import { Post } from '../hooks/posts/model';

export const processPosts = (fetchedPosts: Post[]): Post[] => {
  const posts = fetchedPosts.map((post) => {
    const categoryId = post.categories.length > 0 ? post.categories[0] : null;
    const topicId = post.topic.length > 0 ? post.topic[0] : null;
    let categoryName;
    let topicName;
    // eslint-disable-next-line no-underscore-dangle
    post._embedded['wp:term'].forEach((value) => {
      const item = value.length > 0 ? value[0] : null;
      if (item != null) {
        if (item.id === categoryId) {
          categoryName = item.name;
        }
        if (item.id === topicId) {
          topicName = item.name;
        }
      }
    });
    return { ...post, categoryName, topicName };
  });
  return posts;
};

export const isoToLocal = (time: string) => (
  DateTime.fromISO(time).toLocal().toFormat('dd MMMM yyyy')
);
