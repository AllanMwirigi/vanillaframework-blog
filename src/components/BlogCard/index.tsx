/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import './styles.scss';

function BlogCard() {
  return (
    <div className="col-4">
      <div className="p-card blog-post-header">
        <header>
          <h4>CLOUD AND SERVER</h4>
        </header>
        <hr className="header-break" />
        <img
          className="p-card__image"
          src="https://assets.ubuntu.com/v1/0f33d832-The-State-of-Robotics.jpg"
          alt="Blog Post"
        />
        <div className="p-card__inner">
          <h3>
            <a href="#">The power of installed based snap metrics</a>
          </h3>
          <p>
            <em>
              by
              {' '}
              <a href="#">Bartek Szopka</a>
              {' '}
              on 21st August 2021
            </em>
          </p>
        </div>
        <hr className="u-no-margin--bottom" />
        <div className="p-card__inner">
          Article
        </div>
      </div>
    </div>
  );
}
export default BlogCard;
