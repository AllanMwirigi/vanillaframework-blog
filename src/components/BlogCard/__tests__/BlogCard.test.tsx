import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import BlogCard from '..';
import { processedPosts } from '../../../utils/mock-data';

describe('BlogCard', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render post category', () => {
    render(<BlogCard post={processedPosts[0]} />);
    const heading = screen.getByText('People and culture', { exact: false });
    expect(heading).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<BlogCard post={processedPosts[0]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
