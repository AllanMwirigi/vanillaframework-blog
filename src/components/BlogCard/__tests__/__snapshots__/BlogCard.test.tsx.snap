// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard matches snapshot 1`] = `
<div
  className="col-4 blog-post-col"
>
  <div
    className="p-card--highlighted blog-post"
  >
    <header>
      <h4
        className="p-muted-heading"
      >
        People and culture
      </h4>
    </header>
    <hr
      className="header-break"
    />
    <div
      className="p-card__content"
    >
      <img
        alt="Blog Post"
        className="p-card__image"
        src="https://ubuntu.com/wp-content/uploads/5288/Title-Cards.jpg"
      />
      <div>
        <h4>
          <a
            href="https://ubuntu.com/blog/travel-clis-and-sticky-notes-lilyanas-life-as-a-canonical-ux-designer/"
            rel="noreferrer"
            target="_blank"
          >
            Travel, CLIs, and sticky notes: Lilyanaâ€™s life as a Canonical UX designer
          </a>
        </h4>
        <p>
          <em>
            By
             
            <a
              href="https://ubuntu.com/blog/author/cmoullec/"
              rel="noreferrer"
              target="_blank"
            >
              cmoullec
            </a>
             
            on
             
            25 August 2020
          </em>
        </p>
      </div>
    </div>
    <div
      className="blog-card-footer"
    >
      <span
        className="p-text--small"
      >
        Articles
      </span>
    </div>
  </div>
</div>
`;
